Task 2 suntsov

Waltans opened the pull request.

totheskysx10 committed:
Исправление замечаний

totheskysx10 committed:
Исправление названия переменной в passwordService

Waltans committed:
Правки

Waltans committed:
логи

Waltans committed:
логи

Waltans committed:
просто мр

totheskysx10 commented on src/main/java/ru/naumen/service/CommandService.java:
```diff
@@ -34,45 +44,149 @@ public CommandService(EncodeService encodeService, PasswordService passwordServi
+            List<String> params = totalUserParams.get(userId);
+            params.add(complexity);
+            totalUserState.put(userId, nextState);
+            if (nextState == null) {
```
поменяй null на NONE, из-за этого падает

Waltans replied:
Ага, не проверил генерацию, подумал лучше уж какое-то состояние, чем null

totheskysx10 commented on src/main/java/ru/naumen/bot/TelegramBot.java:
```diff
@@ -62,34 +62,39 @@ private void sendMessageToChat(String message, String id) {
-        keyboardRowSecond.add(new KeyboardButton("Помощь"));
+        keyboardRowSecond.add(new KeyboardButton(Command.DELETE_KEYBOARD));
+        keyboardRowSecond.add(new KeyboardButton(Command.EDIT_KEYBOARD));
+        keyboardRowSecond.add(new KeyboardButton(Command.HELP));
```
Command.HELP_KEYBOARD, а не просто хелп)

totheskysx10 commented on src/main/java/ru/naumen/bot/Command.java:
```diff
@@ -8,25 +8,39 @@
  */
 public class Command {
     public static final String GENERATE = "/generate";
+    public static final String GENERATE_KEYBOARD = "Генерировать пароль";
```
сделай вместо "Генерировать пароль" просто "Генерировать", тогда не придётся валидацию переделывать под многословные команды)

Waltans replied:
Я еще не придумал как можно хорошо можно смаппить кнопки и команды, может у тебя есть идеи?

totheskysx10 replied:
В целом вариант с отображением <команда, текстКнопки> меня устраивает)

totheskysx10 reviewed (commented):
Написал правки какие увидел)

Waltans committed:
Доработал сервисы:

- CommandService теперь возвращает Response с состоянием бота
- Добавил клавиатуру для выбора сложности пароля
- Добавил JavaDoc
- Исправил замечания
_ Поправил несколько багов

Waltans committed:
ТЕСТЫ

totheskysx10 commented on src/main/java/ru/naumen/service/CommandService.java:
```diff
@@ -63,45 +63,99 @@ public String performCommand(String message, long userId, String username) {
             case Command.LIST, Command.LIST_KEYBOARD -> getUserPasswords(userId);
             case Command.DELETE, Command.DELETE_KEYBOARD -> deletePassword(splitCommand, userId);
             case Command.EDIT, Command.EDIT_KEYBOARD -> updatePassword(splitCommand, userId);
```
при исполнении команды сначала он спрашивает тебя все данные, а уже когда ты всё ввел, только потом проверяет есть ли пароль с указанным индексом

Waltans replied:
исправил

Waltans committed:
исправил замечания

totheskysx10 reviewed (approved):
OK!
